arch/xtensa/include/kernel_arch_func.h

```diff
@@ -75,6 +75,7 @@ static ALWAYS_INLINE void arch_cohere_stacks(struct k_thread *old_thread,
 					     void *old_switch_handle,
 					     struct k_thread *new_thread)
 {
+	/* KERNEL_COHERENCE selects THREAD_STACK_INFO */
```

Comment:
Yes it does, and that's documented in the Kconfig where it happens.   In general there's no need to call out "Hey!  I'm using an optional API!", you just use it.  Don't think this comment is needed, but not a big deal.